File: backend/src/models/Notificacion.js
const { DataTypes } = require("sequelize");
const sequelize = require("../config/database");

const Notificacion = sequelize.define(
	"Notificacion",
	{
		id_notificacion: {
			type: DataTypes.INTEGER,
			primaryKey: true,
			autoIncrement: true,
		},
		id_empresa: {
			type: DataTypes.INTEGER,
			allowNull: false,
			references: {
				model: "empresas",
				key: "id_empresa",
			},
		},
		id_usuario: {
			type: DataTypes.INTEGER,
			references: {
				model: "usuarios",
				key: "id_usuario",
			},
		},
		tipo: {
			type: DataTypes.ENUM(
				"STOCK_BAJO",
				"STOCK_AGOTADO",
				"VENTA",
				"COMPRA",
				"SISTEMA",
			),
			allowNull: false,
		},
		titulo: {
			type: DataTypes.STRING(200),
			allowNull: false,
		},
		mensaje: {
			type: DataTypes.TEXT,
			allowNull: false,
		},
		leida: {
			type: DataTypes.BOOLEAN,
			defaultValue: false,
		},
		fecha_creacion: {
			type: DataTypes.DATE,
			defaultValue: DataTypes.NOW,
		},
	},
	{
		tableName: "notificaciones",
		timestamps: false,
	},
);

module.exports = Notificacion;

File: mobile/lib/models/notificacion.dart
class Notificacion {
  final int id;
  final String tipo;
  final String titulo;
  final String mensaje;
  final bool leida;
  final DateTime fecha;
  final Map<String, dynamic>? datos;

  Notificacion({
    required this.id,
    required this.tipo,
    required this.titulo,
    required this.mensaje,
    required this.leida,
    required this.fecha,
    this.datos,
  });

  factory Notificacion.fromJson(Map<String, dynamic> json) {
    return Notificacion(
      id: json['id_notificacion'] ?? json['id'],
      tipo: json['tipo'] ?? '',
      titulo: json['titulo'] ?? '',
      mensaje: json['mensaje'] ?? '',
      leida: json['leida'] == 1 || json['leida'] == true,
      fecha: DateTime.parse(json['fecha_creacion'] ?? json['fecha']),
      datos: json['datos'],
    );
  }

  // Tipos de notificación según la base de datos
  bool get isStockBajo => tipo == 'STOCK_BAJO';
  bool get isStockAgotado => tipo == 'STOCK_AGOTADO';
  bool get isVenta => tipo == 'VENTA';
  bool get isCompra => tipo == 'COMPRA';
  bool get isSistema => tipo == 'SISTEMA';
}
